### ISSUE TITLE
fix import test

### ISSUE THREAD

**Issue by efiop** [iterative/dvc-gs#7]
https://github.com/iterative/dvc-gs/runs/7830837079?check_suite_focus=true

**Comment by dtrifiro** [comment]
Test is broken because of https://github.com/fsspec/gcsfs/pull/488.

This is what the gcsfs dircache looks like: 
```json
{
  "daniele-dvc-testing/cf8f2fbe-89e5-4871-a71d-ac24042de128/dir": [
    "daniele-dvc-testing/cf8f2fbe-89e5-4871-a71d-ac24042de128/dir/file",
    "daniele-dvc-testing/cf8f2fbe-89e5-4871-a71d-ac24042de128/dir/subdir/subfile"
  ],
  "daniele-dvc-testing/cf8f2fbe-89e5-4871-a71d-ac24042de128/dir/subdir": [
    "daniele-dvc-testing/cf8f2fbe-89e5-4871-a71d-ac24042de128/dir/subdir/subfile"
  ]
}
```

Failure is due to the fact that `ls_from_cache` returns `subfile` as a file in `dir`, even though it's contained in `dir/subdir`

**Comment by efiop** [comment]
@dtrifiro Thanks for the research! 🙏

**Comment by efiop** [comment]
@dtrifiro Btw, so you've tried with that patch and it works, right?

**Comment by dtrifiro** [comment]
It doesn't raise `FileNotFoundError` on `_ls_from_cache` anymore, but we have other issues: each time we run the test, the stage's md5 and the deps' dir md5 changes.

The reason is that we're using `etag` instead of `md5` in `HashInfo` when saving/building the stage's deps. Since the `etag` changes on every test run (new objects/etags each time, after all), the dependency `.dir` hash changes too, which means the test fails because both the dir's and stage's `md5` change.

Setting `stage_md5` and `dir_md5` to `None` here
https://github.com/iterative/dvc-gs/blob/33d6f2f9731b0d37e4473fb9db17c239099d9916/dvc_gs/tests/test_dvc.py#L19-L30

would make the test pass, although this is not the best solution, as we could be asserting on the `outs` section although I think we need to revisit the usage of `stage_md5` and `dir_md5` in `dvc/testing/test_workspace.py::TestImport::test_import_dir`.

Something like the following might work, although it does not check the deps' and the stage's md5

```diff
diff --git a/dvc/testing/test_workspace.py b/dvc/testing/test_workspace.py
index 0b883a7ba..3b651b9a7 100644
 --- a/dvc/testing/test_workspace.py 
 +++ b/dvc/testing/test_workspace.py
@@ -19,7 +19,7 @@ class TestImport:
     def dir_md5(self):
         pytest.skip()
 
-    def test_import_dir(self, tmp_dir, dvc, workspace, stage_md5, dir_md5):
+    def test_import_dir(self, tmp_dir, dvc, workspace):
         from dvc.odbmgr import ODBManager
 
         workspace.gen(
@@ -33,7 +33,7 @@ class TestImport:
         dvc.odb = ODBManager(dvc)
 
         assert not (tmp_dir / "dir").exists()  # sanity check
-        dvc.imp_url("remote://workspace/dir")
+        imported = dvc.imp_url("remote://workspace/dir")
         assert set(os.listdir(tmp_dir / "dir")) == {"file", "subdir"}
         assert (tmp_dir / "dir" / "file").read_text() == "file"
         assert list(os.listdir(tmp_dir / "dir" / "subdir")) == ["subfile"]
@@ -43,21 +43,19 @@ class TestImport:
 
         assert dvc.status() == {}
 
-        if stage_md5 is not None and dir_md5 is not None:
-            assert (tmp_dir / "dir.dvc").read_text() == (
-                f"md5: {stage_md5}\n"
-                "frozen: true\n"
-                "deps:\n"
-                f"- md5: {dir_md5}\n"
-                "  size: 11\n"
-                "  nfiles: 2\n"
-                "  path: remote://workspace/dir\n"
-                "outs:\n"
-                "- md5: b6dcab6ccd17ca0a8bf4a215a37d14cc.dir\n"
-                "  size: 11\n"
-                "  nfiles: 2\n"
-                "  path: dir\n"
-            )
+        assert (tmp_dir / "dir.dvc").exists()
+        assert imported.frozen
+        assert imported.outs[0].hash_info.name == "md5"
+        assert (
+            imported.outs[0].hash_info.value
+            == "b6dcab6ccd17ca0a8bf4a215a37d14cc.dir"
+        )
+        assert imported.deps[0].meta.size == 11
+        assert imported.deps[0].meta.nfiles == 2
+        assert imported.deps[0].def_path == "remote://workspace/dir"
+
+        assert imported.outs[0].meta.size == 11
+        assert imported.outs[0].meta.nfiles == 2
 
     @pytest.fixture
     def is_object_storage(self):
```

**Comment by efiop** [comment]
@dtrifiro Ah, right, that's why those tests weren't enabled in the first place, I completely forgot. If setting None (or fixing this test on dvc-gs side in any other way) works - it is good enough for now. Thanks for the research!

**Comment by efiop** [comment]
Fixed by #9 Thank you @dtrifiro 🙏
